app/lib/frontend/keys.dart

```diff
@@ -24,3 +24,19 @@ Future<String> customSearchKeyFromDB() async {
 
   return apiKey.value;
 }
+
+String _cachedNotificationSecret;
+
+/// Gets the shared notification secret from the datastore (or local cache).
+///
+/// Ignoring cache invalidations for now, because at worst the notification is
+/// not acknowledged, and the related processing will happen 10 minutes later.
+Future<String> getNotificationSecret() async {
+  if (_cachedNotificationSecret == null) {
+    final db = dbService;
+    final key = db.emptyKey.append(PrivateKey, id: 'notification-secret');
+    final PrivateKey privateKey = (await db.lookup([key])).first;
+    _cachedNotificationSecret = privateKey?.value ?? 'secret';
```

Comment:
We shouldn't default to a constant in the sources. If it's not available it should be `null` and validation should fail.

You can disable validation code (make it always return `true`) until the prod version has the secret set and enable the validation afterwards.